consumer/consumer.go

```diff
@@ -70,39 +76,31 @@ func (c *Consumer) setup() {
 	}
 }
 
-func newClusterConfig(clientID string) *cluster.Config {
-	c := cluster.NewConfig()
-	c.ClientID = clientID
-	c.Consumer.Return.Errors = true
-	// Specify that we are using at least Kafka v1.0
-	c.Version = sarama.V1_0_0_0
-	// Distribute load across instances using round robin strategy
-	c.Group.PartitionStrategy = cluster.StrategyRoundRobin
-	// One chan per partition instead of default multiplexing behaviour.
-	c.Group.Mode = cluster.ConsumerModePartitions
-
-	return c
-}
-
 // Serve runs the consumer and listens for new messages on the given
 // topics.  You must provide it with unique clientID and the address
 // of one or more Kafka brokers.  Serve will block until it is
 // instructed to stop, which you can achieve by calling Consumer.Stop.
 // When Serve terminates it will return an Error or nil to indicate
 // that it excited without error.
-func (c *Consumer) Serve(clientID string, addrs ...string) error {
+func (c *Consumer) Serve(config Config, addrs ...string) error {
 	c.setup()
 
-	c.config = newClusterConfig(clientID)
+	c.config = &config
+	// Always make sure we are using the right group mode: One chan per partition instead of default multiplexing behaviour.
+	if c.config.Group.Mode != cluster.ConsumerModePartitions {
```

Comment:
I would prefer to see this check as a separate function, with a test (separating it out from `Serve` should make that trivial.  Probably call it something like `validateConfig` so we can add further checks there.